content/storage/bind-mounts.md

```diff
@@ -266,6 +266,39 @@ $ docker container stop devtest
 $ docker container rm devtest
 ```
 
+## Recursive mounts
+
+When you bind mount a path that itself contains mounts, those submounts are
+also included in the bind mount by default. This behavior is configurable,
+using the `bind-recursive` option for `--mount`.
+
+If the bind mount is read-only, the Docker Engine makes a best-effort attempt
+at making the submounts read-only as well. This is referred to as recursive
+read-only mounts. Recursive read-only mounts require Linux kernel version 5.12
+or later. If you're running an older kernel version, submounts are
+automatically mounted as read-write by default. Attempting to set submounts to
+be read-only on a kernel version earlier than 5.12, using the
+`bind-recursive=readonly` option, results in an error.
+
+Supported values for the `bind-recursive` option are:
+
+`bind-recursive=<enabled|true|1>` (default)
```

Comment:
Should we use a table for this list of options? I think it may be slightly more readable;


```markdown
| Value               | Description                                                                                                       |
|:--------------------|:------------------------------------------------------------------------------------------------------------------|
| `enabled` (default) | Read-only mounts are made recursively read-only if kernel is v5.12 or later. Otherwise, submounts are read-write. |
| `disabled`          | Submounts are ignored (not included in the bind mount).                                                           |
| `writable`          | Submounts are read-write.                                                                                         |
| `readonly`          | Submounts are read-only. Requires kernel v5.12 or later.                                                          |
```

I wasn't sure how to best mention the alternative (true/false 1/0) boolean format; perhaps something like;

```markdown
| Value               | Description                                                                                                                                                                          |
|:--------------------|:-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| `enabled` (default) | Read-only mounts are made recursively read-only if kernel is v5.12 or later. Otherwise, submounts are read-write. For convenience, boolean values ("true" or "1") are also accepted. |
| `disabled`          | Submounts are ignored (not included in the bind mount). For convenience, boolean values ("false" or "0") are also accepted.                                                          |
| `writable`          | Submounts are read-write.                                                                                                                                                            |
| `readonly`          | Submounts are read-only. Requires kernel v5.12 or later.                                                                                                                             |
```

Reply:
I'm actually considering now if we should keep the `true`, `false`, `1`, `0`. We supported this for the (now soft-deprecated) `bind-nonrecursive` option, but that only supported a boolean.

The boolean values for the new option would _slightly_ help users to transition, but we don't document that in the deprecation warning;

```
bind-nonrecursive is deprecated, use bind-recursive=disabled instead
```

So perhaps we don't really need it, and having the extra possibilities may only be adding more confusion. If there's strong reason to add them, adding them later won't be too complicated.

I'll open a PR for discussion

Reply:
- opened https://github.com/docker/cli/pull/4671